src/libtriton/bindings/python/objects/pyTritonContext.cpp

```diff
@@ -1265,6 +1266,9 @@ namespace triton {
         catch (const triton::exceptions::Exception& e) {
           return PyErr_Format(PyExc_TypeError, "%s", e.what());
         }
+        catch (z3::exception e) {
+          std::cerr << "PyObject* TritonContext_getModel(): z3::exception: " << e.msg() << std::endl;
+        }
```

Comment:
I think it is better to add this exception into `z3Solver.cpp` instead of the bindings part.

Reply:
I think so too. But, strange to say, I cannot catch z3::exception in SolverEngine.cpp (see attached patch file).
Is there any my mistake?

I also attached python script triggers z3::exception.

[handling-z3-exception-in-solverEngine-getModels.patch.txt](https://github.com/JonathanSalwan/Triton/files/2130928/handling-z3-exception-in-solverEngine-getModels.patch.txt)
[z3-exception.py.txt](https://github.com/JonathanSalwan/Triton/files/2130929/z3-exception.py.txt)